File: typescript/parser/src/parser/json/jbool-parser.ts
import { JBool } from './type/jbool'
import { JParser } from './jparser'
import { ParseResult, ParseSuccess, ParseFailer } from '../common/parser-result'
import * as P from '../common/parser'

class JBoolParser implements JParser<JBool> {
	parser = P.or(P.string("true"), P.string("false"))

	parse(input: string): ParseResult<JBool> {
		const result = this.parser.parse(input)
		if (result instanceof ParseSuccess) {
			if(result.value == "true") {
				return new ParseSuccess(new JBool(true), result.value)
			} else {
				return new ParseSuccess(new JBool(false), result.value)
			}
		} else if (result instanceof ParseFailer) {
			return new ParseFailer(result.message, result.next)
		}
	}
}
export const jBoolParser = new JBoolParser()

File: scala/parsers/src/main/scala/jp/co/teruuu/parser/json/JBoolParser.scala
package jp.co.teruuu.parser.json

import jp.co.teruuu.parser.common.{ParseNg, ParseOk, ParseResult, Parser}

object JBoolParser extends JParser[JBool] {

  override def parse(input: String): ParseResult[JBool] = {
    val parser = Parser.string("true").or(Parser.string("false"))
    parser.parse(input) match {
      case result: ParseOk[String] => ParseOk(JBool(result.value.equals("true")), result.next)
      case result: ParseNg[_] => ParseNg(result.message, result.next)
    }
  }
}
